src/software/new_geom/vector.cpp

```diff
@@ -51,7 +51,10 @@ double Vector::length() const
 
 Vector Vector::normalize() const
 {
-    return length() < 1.0e-9 ? Vector() : Vector(x_ / length(), y_ / length());
+    // 2 * FIXED_EPSILON for error needed:
+    // - mathcalls `hypot` from `length` function call
+    // - where FIXED_EPSILON is used to initialize the vector
```

Comment:
not sure if this comment is consistent with the code below

Reply:
It's referring to if someone initializes a vector using FIXED_EPSILON to provide a near-zero value. That's why I think we need 2 * FIXED_EPSILON as opposed to just FIXED_EPSILON and is one of the test-cases.
The only operation below in the check is with `length()` so I'm not sure where the extra error could be coming from otherwise.

Reply:
I see